test/scripts/e2e_subs/app-inner-calls-csp.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+import os
+import sys
+from goal import Goal
+import algosdk.logic as logic
+
+from datetime import datetime
+
+stamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+print(f"{os.path.basename(sys.argv[0])} start {stamp}")
+
+goal = Goal(sys.argv[1], autosend=True)
+
+joe = goal.new_account()
+
+txinfo1, err = goal.pay(goal.account, joe, amt=500_000)
+assert not err, err
+
+txinfo1, err = goal.keyreg(joe, nonpart=True)
+assert not err, err
+joeb = goal.balance(joe)
+
+# app1 calls the clear state program of app2 (after opting into it) which issues an
+# inner app call to app3. This verifies that both accessing a CSP through inner app
+# calls and issuing inner app calls from a CSP is possible.
+app1 = """
+#pragma version 6
+ txn ApplicationID
+ bz end
+
+ itxn_begin
+  int appl
+  itxn_field TypeEnum
+
+  txn Applications 1
+  dup
+  store 0
+  itxn_field ApplicationID
+
+  int OptIn
+  itxn_field OnCompletion
+
+  txn Applications 2
+  itxn_field Applications
+
+ itxn_next
+  int appl
+  itxn_field TypeEnum
+
+  load 0
+  itxn_field ApplicationID
+
+  txn Applications 2
+  itxn_field Applications
+
+  int ClearState
+  itxn_field OnCompletion
+
+ itxn_submit
+
+
+end:
+ int 1
+"""
+
+app2 = """
+#pragma version 6
+ txn ApplicationID
+ bz end
+
+ itxn_begin
+  int appl
+  itxn_field TypeEnum
+
+  txn Applications 1
+  itxn_field ApplicationID
+ itxn_submit
+
+
+ end:
+  int 1
+"""
+
+app3 = """
+#pragma version 6
+int 1
+"""
+
+goal.autosend = True
+
+txinfo1, err = goal.app_create(joe, goal.assemble(app1))
```

Comment:
@algoidurovic and others - For illustrative purposes, I experimented with removing code duplication stemming from repeated `app_create ` invocations:  https://github.com/algoidurovic/go-algorand/pull/1.

While I don't expect to merge the PR and will happily close it, I provide it in case others find it interesting.  If nothing else, creating the PR helped me learn about Python.